### DISCUSSION TITLE
GZip while building the Gateway app

### DISCUSSION THREAD

**Discussion by alvaro-ricotta** [bobanetwork/boba_legacy#1201] [Ideas]
Currently, our bundle exceeds 15MB, which makes the load time slow, especially for mobile devices or slow connections.

The code can be optimized using gzip/dgzip compression with results up to 15 times better. Attached is a reference link:
https://codetoart.com/blog/how-to-load-react-web-app-faster

@sk-enya @InoMurko @wsdt @souradeep-das  Any other idea, recommendation, or a better alternative is welcome.

**Comment by wsdt** [comment]
1. Do we already use a Bundle analyzer, to see what is the cause of the bundle size? 
-> maybe there is a huge library included that can be easily replaced for example. 

2. We could split the bundle into chunks, the other chunks are then loaded on a on-demand basis. 

3. Lazy loading pattern on the gateway 

4. and yes Gzip is definitely also something we should look into imho. 

Hope this helps.

**Reply by alvaro-ricotta** [reply]
1. We gradually introduced specific changes in the way we do things, especially in the import and export of external libraries and how we use them within our Gateway. Obviously, there are many things still pending for improvement, and we will address them. To start, we could use the Webpack Bundle Analyzer.
3. React lazy is an excellent recommendation; I will add it to the backlog, and as soon as we find the time to implement it, we will.
4. I believe this is essential because even if we follow points 1 and 2 perfectly, compression by itself initially does more than the rest.

**Comment by sk-enya** [comment]
Thanks @alvaro-ricotta for raising the discussion. 
Absolutely, as @wsdt said it's good points to follow and it applied earliest possible on gateway.

Let me add those points in gateway 2.5.1 release, so we can investigate fix it asap.

**Comment by sk-enya** [comment]
@alvaro-ricotta @wsdt ,
Just to update here, we are already doing gzip of bundle files. 
![Screenshot 2023-09-14 at 12 26 15 PM](https://github.com/bobanetwork/boba/assets/86316370/ae766e14-1de6-4e30-bf38-b09f9fb92efb)

**Comment by sk-enya** [comment]
Here is the PR which covers.
1. lazy loading on gateway.
https://github.com/bobanetwork/boba/pull/1212

**Comment by sk-enya** [comment]
As we are already covering all the points below on gateway so closing the discussion.

 - [x] Do we already use a Bundle analyzer, to see what is the cause of the bundle size?
-> maybe there is a huge library included that can be easily replaced for example.
 - [x] Lazy loading pattern on the gateway
 - [x] Gzip is definitely also something we should look into imho.
